File: medium/2492_minimum_score_of_a_path_between_two_cities/solution.cpp
#include "../../utils/cpp/union_find.hpp"
#include <vector>

class Solution {
public:
  int minScore(int n, std::vector<std::vector<int>> &roads) {
    utils::UnionFind uf(n + 1);

    for (const std::vector<int> &road : roads) {
      if (uf.find(road[0]) != uf.find(road[1]))
        uf.merge(road[0], road[1]);
    }

    int minDistance = INT_MAX;

    for (const std::vector<int> &road : roads) {
      if (uf.find(1) == uf.find(road[0]))
        minDistance = std::min(minDistance, road[2]);
    }

    return minDistance;
  }
};

File: medium/2492_minimum_score_of_a_path_between_two_cities/solution.py
from utils.python import UnionFind


class Solution:
    def minScore(self, n: int, roads: list[list[int]]) -> int:
        uf = UnionFind(n + 1)

        for a, b, _ in roads:
            if uf.find(a) != uf.find(b):
                uf.union(a, b)

        minDistance = float("inf")

        for a, _, distance in roads:
            if uf.find(1) == uf.find(a):
                minDistance = min(minDistance, distance)

        return minDistance
